parquet-hadoop/pom.xml

```diff
@@ -102,6 +102,11 @@
       <type>jar</type>
       <scope>compile</scope>
     </dependency>
+    <dependency>
+      <groupId>io.airlift</groupId>
```

Comment:
somebody more familiar with parquet-mr, on whether adding tis dependency is OK.

Reply:
Generally, we are strict to add more dependencies. Do we know if this dependency also brings in other dependencies too? What total libraries were introduced and what is their footprint of them?

Reply:
@shangxinli  I have checked that **io.airlift:aircompressor** does not bring any new dependency. Please check the following result:

> ✗ mvn dependency:tree
> [INFO] Scanning for projects...
> [INFO]
> [INFO] -----------------< org.apache.parquet:parquet-hadoop >------------------
> [INFO] Building Apache Parquet Hadoop 1.13.0-SNAPSHOT
> [INFO] --------------------------------[ jar ]---------------------------------
> [INFO]
> [INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ parquet-hadoop ---
> [INFO] org.apache.parquet:parquet-hadoop:jar:1.13.0-SNAPSHOT
> [INFO] +- org.apache.parquet:parquet-column:jar:1.13.0-SNAPSHOT:compile
> [INFO] |  \- org.apache.parquet:parquet-encoding:jar:1.13.0-SNAPSHOT:compile
> [INFO] +- org.apache.parquet:parquet-format-structures:jar:1.13.0-SNAPSHOT:compile
> [INFO] +- org.apache.parquet:parquet-common:jar:1.13.0-SNAPSHOT:compile
> [INFO] +- org.apache.parquet:parquet-column:test-jar:tests:1.13.0-SNAPSHOT:test
> [INFO] +- org.apache.hadoop:hadoop-client:jar:3.2.3:provided
> [INFO] |  +- org.apache.hadoop:hadoop-hdfs-client:jar:3.2.3:provided
> [INFO] |  |  \- com.squareup.okhttp:okhttp:jar:2.7.5:provided
> [INFO] |  +- org.apache.hadoop:hadoop-yarn-api:jar:3.2.3:provided
> [INFO] |  |  \- javax.xml.bind:jaxb-api:jar:2.2.11:provided
> [INFO] |  +- org.apache.hadoop:hadoop-yarn-client:jar:3.2.3:provided
> [INFO] |  \- org.apache.hadoop:hadoop-mapreduce-client-jobclient:jar:3.2.3:provided
> [INFO] |     \- org.apache.hadoop:hadoop-mapreduce-client-common:jar:3.2.3:provided
> [INFO] +- org.apache.hadoop:hadoop-common:jar:3.2.3:provided
> [INFO] |  +- commons-cli:commons-cli:jar:1.2:provided
> [INFO] |  +- org.apache.commons:commons-math3:jar:3.1.1:provided
> [INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.13:provided
> [INFO] |  |  \- org.apache.httpcomponents:httpcore:jar:4.4.13:provided
> [INFO] |  +- commons-codec:commons-codec:jar:1.11:provided
> [INFO] |  +- commons-net:commons-net:jar:3.6:provided
> [INFO] |  +- commons-collections:commons-collections:jar:3.2.2:provided
> [INFO] |  +- javax.servlet:javax.servlet-api:jar:3.1.0:provided
> [INFO] |  +- javax.activation:javax.activation-api:jar:1.2.0:provided
> [INFO] |  +- org.eclipse.jetty:jetty-server:jar:9.4.40.v20210413:provided
> [INFO] |  |  +- org.eclipse.jetty:jetty-http:jar:9.4.40.v20210413:provided
> [INFO] |  |  \- org.eclipse.jetty:jetty-io:jar:9.4.40.v20210413:provided
> [INFO] |  +- org.eclipse.jetty:jetty-util:jar:9.4.40.v20210413:provided
> [INFO] |  +- org.eclipse.jetty:jetty-servlet:jar:9.4.40.v20210413:provided
> [INFO] |  |  +- org.eclipse.jetty:jetty-security:jar:9.4.40.v20210413:provided
> [INFO] |  |  \- org.eclipse.jetty:jetty-util-ajax:jar:9.4.40.v20210413:provided
> [INFO] |  +- org.eclipse.jetty:jetty-webapp:jar:9.4.40.v20210413:provided
> [INFO] |  |  \- org.eclipse.jetty:jetty-xml:jar:9.4.40.v20210413:provided
> [INFO] |  +- javax.servlet.jsp:jsp-api:jar:2.1:provided
> [INFO] |  +- com.sun.jersey:jersey-core:jar:1.19:provided
> [INFO] |  |  \- javax.ws.rs:jsr311-api:jar:1.1.1:provided
> [INFO] |  +- com.sun.jersey:jersey-servlet:jar:1.19:provided
> [INFO] |  +- com.sun.jersey:jersey-json:jar:1.19:provided
> [INFO] |  |  +- org.codehaus.jettison:jettison:jar:1.1:provided
> [INFO] |  |  +- com.sun.xml.bind:jaxb-impl:jar:2.2.3-1:provided
> [INFO] |  |  +- org.codehaus.jackson:jackson-core-asl:jar:1.9.2:provided
> [INFO] |  |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.2:provided
> [INFO] |  |  +- org.codehaus.jackson:jackson-jaxrs:jar:1.9.2:provided
> [INFO] |  |  \- org.codehaus.jackson:jackson-xc:jar:1.9.2:provided
> [INFO] |  +- com.sun.jersey:jersey-server:jar:1.19:provided
> [INFO] |  +- commons-logging:commons-logging:jar:1.1.3:provided
> [INFO] |  +- log4j:log4j:jar:1.2.17:provided
> [INFO] |  +- commons-beanutils:commons-beanutils:jar:1.9.4:provided
> [INFO] |  +- org.apache.commons:commons-configuration2:jar:2.1.1:provided
> [INFO] |  +- org.apache.commons:commons-text:jar:1.4:provided
> [INFO] |  +- org.apache.avro:avro:jar:1.7.7:provided
> [INFO] |  |  \- com.thoughtworks.paranamer:paranamer:jar:2.3:provided
> [INFO] |  +- com.google.re2j:re2j:jar:1.1:provided
> [INFO] |  +- com.google.protobuf:protobuf-java:jar:2.5.0:provided
> [INFO] |  +- com.google.code.gson:gson:jar:2.2.4:provided
> [INFO] |  +- org.apache.hadoop:hadoop-auth:jar:3.2.3:provided
> [INFO] |  |  +- com.nimbusds:nimbus-jose-jwt:jar:9.8.1:provided
> [INFO] |  |  |  \- com.github.stephenc.jcip:jcip-annotations:jar:1.0-1:provided
> [INFO] |  |  +- net.minidev:json-smart:jar:2.4.7:provided
> [INFO] |  |  |  \- net.minidev:accessors-smart:jar:2.4.7:provided
> [INFO] |  |  |     \- org.ow2.asm:asm:jar:9.1:provided
> [INFO] |  |  \- org.apache.curator:curator-framework:jar:2.13.0:provided
> [INFO] |  +- com.jcraft:jsch:jar:0.1.55:provided
> [INFO] |  +- org.apache.curator:curator-client:jar:2.13.0:provided
> [INFO] |  +- org.apache.curator:curator-recipes:jar:2.13.0:provided
> [INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:provided
> [INFO] |  +- org.apache.htrace:htrace-core4:jar:4.1.0-incubating:provided
> [INFO] |  +- org.apache.zookeeper:zookeeper:jar:3.4.14:provided
> [INFO] |  |  +- com.github.spotbugs:spotbugs-annotations:jar:3.1.9:provided
> [INFO] |  |  \- jline:jline:jar:0.9.94:provided
> [INFO] |  +- org.apache.commons:commons-compress:jar:1.21:provided
> [INFO] |  +- org.apache.kerby:kerb-simplekdc:jar:1.0.1:provided
> [INFO] |  |  +- org.apache.kerby:kerb-client:jar:1.0.1:provided
> [INFO] |  |  |  +- org.apache.kerby:kerby-config:jar:1.0.1:provided
> [INFO] |  |  |  +- org.apache.kerby:kerb-core:jar:1.0.1:provided
> [INFO] |  |  |  |  \- org.apache.kerby:kerby-pkix:jar:1.0.1:provided
> [INFO] |  |  |  |     +- org.apache.kerby:kerby-asn1:jar:1.0.1:provided
> [INFO] |  |  |  |     \- org.apache.kerby:kerby-util:jar:1.0.1:provided
> [INFO] |  |  |  +- org.apache.kerby:kerb-common:jar:1.0.1:provided
> [INFO] |  |  |  |  \- org.apache.kerby:kerb-crypto:jar:1.0.1:provided
> [INFO] |  |  |  +- org.apache.kerby:kerb-util:jar:1.0.1:provided
> [INFO] |  |  |  \- org.apache.kerby:token-provider:jar:1.0.1:provided
> [INFO] |  |  \- org.apache.kerby:kerb-admin:jar:1.0.1:provided
> [INFO] |  |     +- org.apache.kerby:kerb-server:jar:1.0.1:provided
> [INFO] |  |     |  \- org.apache.kerby:kerb-identity:jar:1.0.1:provided
> [INFO] |  |     \- org.apache.kerby:kerby-xdr:jar:1.0.1:provided
> [INFO] |  +- org.codehaus.woodstox:stax2-api:jar:4.2.1:provided
> [INFO] |  +- com.fasterxml.woodstox:woodstox-core:jar:5.3.0:provided
> [INFO] |  \- dnsjava:dnsjava:jar:2.1.7:provided
> [INFO] +- org.apache.hadoop:hadoop-mapreduce-client-core:jar:3.2.3:provided
> [INFO] |  +- org.apache.hadoop:hadoop-yarn-common:jar:3.2.3:provided
> [INFO] |  |  +- com.sun.jersey:jersey-client:jar:1.19:provided
> [INFO] |  |  +- com.google.inject:guice:jar:4.0:provided
> [INFO] |  |  |  +- javax.inject:javax.inject:jar:1:provided
> [INFO] |  |  |  \- aopalliance:aopalliance:jar:1.0:provided
> [INFO] |  |  +- com.sun.jersey.contribs:jersey-guice:jar:1.19:provided
> [INFO] |  |  +- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.10.5:provided
> [INFO] |  |  |  +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:provided
> [INFO] |  |  |  \- jakarta.activation:jakarta.activation-api:jar:1.2.1:provided
> [INFO] |  |  \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.10.5:provided
> [INFO] |  |     \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.10.5:provided
> [INFO] |  +- com.google.inject.extensions:guice-servlet:jar:4.0:provided
> [INFO] |  \- io.netty:netty:jar:3.10.6.Final:provided
> [INFO] +- org.apache.hadoop:hadoop-annotations:jar:3.2.3:provided
> [INFO] |  \- jdk.tools:jdk.tools:jar:1.8:system
> [INFO] +- org.apache.parquet:parquet-jackson:jar:1.13.0-SNAPSHOT:runtime
> [INFO] +- com.fasterxml.jackson.core:jackson-core:jar:2.13.2:compile
> [INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
> [INFO] |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.13.2:compile
> [INFO] +- org.xerial.snappy:snappy-java:jar:1.1.8.3:compile
> [INFO] +- io.airlift:aircompressor:jar:0.21:compile
> [INFO] +- commons-pool:commons-pool:jar:1.6:compile
> [INFO] +- it.unimi.dsi:fastutil:jar:8.4.2:compile
> [INFO] +- com.github.luben:zstd-jni:jar:1.5.0-1:compile
> [INFO] +- com.google.guava:guava:jar:27.0.1-jre:test
> [INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:test
> [INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:test
> [INFO] |  +- org.checkerframework:checker-qual:jar:2.5.2:test
> [INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.2.0:test
> [INFO] |  +- com.google.j2objc:j2objc-annotations:jar:1.1:test
> [INFO] |  \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.17:test
> [INFO] +- org.mockito:mockito-all:jar:1.10.19:test
> [INFO] +- org.slf4j:slf4j-simple:jar:1.7.22:test
> [INFO] +- com.squareup.okhttp3:okhttp:jar:4.6.0:test
> [INFO] |  +- com.squareup.okio:okio:jar:2.6.0:provided
> [INFO] |  |  \- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.3.70:provided
> [INFO] |  \- org.jetbrains.kotlin:kotlin-stdlib:jar:1.3.71:provided
> [INFO] |     \- org.jetbrains:annotations:jar:13.0:provided
> [INFO] +- org.slf4j:slf4j-api:jar:1.7.22:compile
> [INFO] +- commons-io:commons-io:jar:2.7:test
> [INFO] +- net.openhft:zero-allocation-hashing:jar:0.9:test
> [INFO] +- org.apache.yetus:audience-annotations:jar:0.13.0:compile
> [INFO] +- org.apache.commons:commons-lang3:jar:3.9:test
> [INFO] +- junit:junit:jar:4.13.1:test
> [INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
> [INFO] \- org.easymock:easymock:jar:3.4:test
> [INFO]    \- org.objenesis:objenesis:jar:2.2:test
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESS
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  0.833 s
> [INFO] Finished at: 2022-10-12T10:23:03+08:00
> [INFO] ------------------------------------------------------------------------

Reply:
From the [pom.xml](https://github.com/airlift/aircompressor/blob/master/pom.xml), its dependencies are either **_provided_** or **_test_**.